### PR TITLE
Fix test discovery for plugin example tests (#1572).

### PR DESCRIPTION
**Pull request by pq**
image

🤘 

@stevemessick @devoncarew

### CODE REVIEW

**Comment by brianegan** [comment]
Could this logic be widened a bit further even? I have a few examples in the my example folder, and I'm not sure this would work for that case? E.g:

`/example/redux/test`, `/example/vanilla/test`, etc

**Comment by pq** [comment]
@brianegan : are these projects in github?  Can you send me a pointer?

Also note that this just gets us part of the way there.  See: #1575

**Comment by devoncarew** [comment]
It _should_ work if `/example/vanilla` has it's own pubspec.

**Comment by brianegan** [comment]
Hey hey -- yep! My repo is here: https://github.com/brianegan/flutter_architecture_samples

If you open the `example/inherited_widgets/test/app_state_test.dart` file, for example, and try to run it, I always get this same type of error:

```
Failed to load "/Users/phillywiggins/lab/flutter_architecture_samples/example/inherited_widget/test/app_state_test.dart":
Unable to spawn isolate: The built-in library 'dart:ui' is not available on the stand-alone VM.
'package:flutter_test/src/binding.dart': error: line 7 pos 1: library handler failed
import 'dart:ui' as ui;
```

I assumed this change would fix that, but if this is unrelated / out of scope please let me know and I'll create a separate issue.

File: src/io/flutter/FlutterUtils.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ public static boolean isInTestDir(DartFile file) {
     if (candidate == null) return false;
 
     final String relativePath = root.getRelativePath(candidate);
-    return relativePath != null && relativePath.startsWith("test/");
+    return relativePath != null && (relativePath.startsWith("test/") || relativePath.startsWith("example/test/"));
```
Comment (devoncarew) [inline src/io/flutter/FlutterUtils.java]: We may want to tweak this check to:
- find the closest pubspec.yaml to `file`
- validate that `file` is within a `test` dir that's a sibling of the pubspec

Comment (devoncarew) [reply src/io/flutter/FlutterUtils.java]: It looks like the previous version of `verifyFlutterTestFile()` was doing something very similar to that (looking for the closest containing pubspec.yaml making sure the relative path starts with `test/`), so I'm guess I'm a bit confused about the best fix.

Probably just best to figure out how  to support `test/`, `example/test/`, and `example/foo/test` in a relatively generic way...

Comment (pq) [reply src/io/flutter/FlutterUtils.java]: Absolutely.  I've been thinking some more about this and agree that this would be a more robust approach. 👍

**Comment by pq** [comment]
Thanks for all of the comments!

I've got a more robust method in mind (see #1583) but will land this in the meantime since it at least fixes #1572.

Expect a more complete fix soon! 👍

**Pull request merged**